src/ConnectionManager.h

```diff
@@ -96,6 +96,10 @@ inline void debugMessage(char *_msg, int _debugLevel, bool _timestamp = true, bo
   }
 }
 
+inline unsigned long ConnectionManager::getTime() {
```

Comment:
Is there any good reason this function (and also the next) is inlined? I suppose it`s because it`s implemented in the header file and otherwise there would be **multiple definition** compile errors. I suggest you consider moving the implementation of the functions in a `ConnectionManager.cpp` file.

Reply:
Another thing: If both `WiFiConnectionManager` and `GsmConnectionManager` overwrite `getTime` (which they do) - in which case is this implementation supposed to be called?

Reply:
It's called in the derived implementations. They provide also a fallback implementation.